modules/compatibility/src/test/java/org/apache/ignite/compatibility/spi/MultiVersionClusterTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compatibility.spi;
+
+import org.apache.ignite.compatibility.testframework.junits.IgniteCompatibilityAbstractTest;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.GridCacheAbstractFullApiSelfTest;
+import org.apache.ignite.lang.IgniteInClosure;
+import org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi;
+
+/**
+ * TODO: remove at completion.
+ */
+public class MultiVersionClusterTest extends IgniteCompatibilityAbstractTest {
+    /** */
+    public void testJoinMultiVersionTopologyLocalFirst() throws Exception {
+        try {
+            IgniteEx ignite = startGrid(0);
+
+            assertEquals(1, topologyVersion(ignite));
+
+            startGrid("testMultiVersion", "2.3.0-SNAPSHOT", new PostConfigurationClosure());
+
+            assertEquals(2, topologyVersion(ignite));
+        }
+        finally {
+            stopAllGrids();
+        }
+    }
+
+    /** */
+    public void testJoinMultiVersionTopologyLocalFirst2() throws Exception {
+        try {
+            IgniteEx ignite = startGrid(0);
+
+            assertEquals(1, topologyVersion(ignite));
+
+            startGrid(1, "2.3.0-SNAPSHOT", new PostConfigurationClosure());
```

Comment:
`2.3.0-SNAPSHOT` definitely should be a constant

Reply:
This is dummy test and will be removed.